### PR TITLE
Enchantment and tool work

### PR DESCRIPTION
**Pull request by supersaiyansubtlety**
refactors:
- `Item`+`ItemStack` `#isSuitableFor` -> `#isCorrectForDrops` to match string-derived field name
- ~~`AbstractBlock.Settings`+`AbstractBlock.AbstractBlockState` `#toolRequired` (field) -> `#requiresTool` for consistency with `Settings`' setter method~~
- ~~`AbstractBlock.AbstractBlockState` `#isToolRequired` -> `#doesRequireTool` to follow boolean getter naming conventions with the new field name~~
- `AbstractBlock.Settings` `#requiresTool` (method) -> `#toolRequired` to match the field name

tagged with t: refactor despite being mostly new because `toolRequired` is quite commonly used ~~and it could go the other way (rename `AbstractBlock.Settings` `#requiresTool` (method) -> `#toolRequired`)~~ 
(it did go the other way)

### CODE REVIEW

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping [hunk 1]
```diff
@@ -262,14 +264,18 @@ CLASS net/minecraft/unmapped/C_jzrpycqo net/minecraft/entity/player/PlayerEntity
 		ARG 1 pos
 		ARG 2 facing
 		ARG 3 stack
+	METHOD m_wbynhygu hasSpaceBelow (DDF)Z
+		ARG 1 xOffset
+		ARG 3 zOffset
+		ARG 5 spaceBellow
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/player/PlayerEntity.mapping]:
```suggestion
		ARG 5 spaceBelow
```

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 2]
```diff
@@ -526,10 +530,10 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 		METHOD m_yvamjply getBlock ()Lnet/minecraft/unmapped/C_mmxmpdoq;
 		METHOD m_yylpqepz onExplosion (Lnet/minecraft/unmapped/C_cdctfzbn;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_aahhrzpf;Ljava/util/function/BiConsumer;)V
 			ARG 3 explosion
-			ARG 4 biConsumer
+			ARG 4 dropConsumer
 		METHOD m_zntvpiao getLuminance ()I
 			COMMENT {@return the light level emitted by this block state}
-		METHOD m_zrpxtrfo isToolRequired ()Z
+		METHOD m_zrpxtrfo doesRequireTool ()Z
```
Comment (eristhea) [inline mappings/net/minecraft/block/AbstractBlock.mapping]: don't like this change, I know it fits conventions better but it just sounds worse

**Review by eristhea — COMMENTED** [review]
looks great, just one nitpick!

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 2 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: would you prefer to change `Settings` `#requiresTool` (method) -> `#toolRequired` or leave it as it was, an exception?

Comment (eristhea) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: both work for me, I would personally leave the `is` on it

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: Oh the `Settings` one is a builder setter method, so it should match the field name
I'll update it to all be`toolRequired`/`isToolRequired`

**Review by eristhea — APPROVED** [review]

**Review by moehreag — APPROVED** [review]

**Pull request merged**